plugins/linux-v4l2/v4l2-udev.c

```diff
@@ -126,13 +132,14 @@ static void *udev_event_thread(void *vptr)
 	fd = udev_monitor_get_fd(mon);
 
 	while (os_event_try(udev_event) == EAGAIN) {
-		FD_ZERO(&fds);
-		FD_SET(fd, &fds);
-		tv.tv_sec = 1;
-		tv.tv_usec = 0;
-
-		if (select(fd + 1, &fds, NULL, NULL, &tv) <= 0)
+		struct pollfd fds[2];
+		fds[0].fd = fd;
+		fds[0].events = POLLIN;
+		fds[1].fd = udev_event_fd;
+		fds[1].events = POLLIN;
+		if (poll(fds, 2, 1000) <= 0) {
```

Comment:
Wouldn't it be better to pass `-1` as the timeout to `poll`? Something like this:

```c
struct pollfd fds[2];

fds[0].fd = fd;
fds[0].events = POLLIN;
fds[1].fd = udev_event_fd;
fds[1].events = POLLIN;

if (poll(&pollfd, 1, -1) <= 0)
	continue;
```

Passing `-1` there signals that we don't care how long it takes, it can block potentially forever.

Reply:
in case we screw this up later it  seems prudent to leave the timeout since return of this thread blocks shutdown. The overhead is negligible imho.